File: python/examples/intents.py
from aristech_nlp_client import NlpClient, GetIntentsRequest
from utils import host, auth_token, auth_secret, root_cert, ssl, project_id

client = NlpClient(host=host, ssl=ssl, root_cert=root_cert, auth_token=auth_token, auth_secret=auth_secret)
intents = client.list_intents(GetIntentsRequest(
    project_id=project_id
))
for intent in intents:
    print(intent)

File: rust/examples/intents.rs
mod utils;
use std::error::Error;
use utils::get_tls_options;

use aristech_nlp_client::{get_client, get_intents, nlp_service::GetIntentsRequest};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    // Load environment variables from .env file
    dotenv::dotenv().ok();

    let host = std::env::var("HOST")?;
    let tls_options = get_tls_options()?;
    let mut client = get_client(host, tls_options).await?;

    let intents = get_intents(
        &mut client,
        GetIntentsRequest {
            project_id: std::env::var("PROJECT_ID")?,
            ..GetIntentsRequest::default()
        },
    )
    .await?;
    for intent in intents {
        println!("{:?}", intent);
    }
    Ok(())
}
